cmd/endpoint.go

```diff
@@ -198,9 +200,47 @@ func (endpoints EndpointList) GetString(i int) string {
 	return endpoints[i].String()
 }
 
-// localEndpointsPerf - returns ServerDrivesPerfInfo for only the
+// localEndpointsMemStat - returns ServerMemStatInfo for only the
 // local endpoints from given list of endpoints
-func localEndpointsPerf(endpoints EndpointList) ServerDrivesPerfInfo {
+func localEndpointsMemStat(endpoints EndpointList) ServerMemStatInfo {
+	var memStats []mem.Stats
+	var addr string
+	for _, endpoint := range endpoints {
```

Comment:
Drives correspond to exact number of `endpoints` - CPU / Mem only correspond to exact number of unique nodes in the deployment. For example, 

A 2 node, 4 drive deployment like this 

```
./minio server http://host1/disk1 http://host1/disk2 http://host2/disk1 http://host2/disk2
```

will report Mem info like this 

```json
[{
	"addr": "host1",
	"perf": [{
		"mem": "32.81M"
	}, {
		"mem": "32.81M"
	}]
}, {
	"addr": "host2",
	"perf": [{
		"mem": "28.68M"
	}, {
		"mem": "28.68M"
	}]
}]
```

It should report it like 

```
[{
	"addr": "host1",
	"perf": [{
		"mem": "32.81M"
	}]
}, {
	"addr": "host2",
	"perf": [{
		"mem": "28.68M"
	}]
}]
```

Reply:
This is inaccurate. The CPU and mem stats reported correspond to each endpoint, and not each node. 

I measure the CPU and mem utilization at the process level, not at the node level. If you notice, the ReadMemStats syscall, and the GETTIME syscall, both operate at the process level. The former obtains the  memory allocated to the process (heap, stack, RO mmapped pages etc.), and the latter measures the amount of clock ticks used by the minio process in a given window.

Reply:
No, you’re right. Sorry, marked it as resolved by mistake. I’ll fix this.

Reply:
Hmmm, I just ran a two node 4 disk cluster with the existing code. The endpoint computation only seems to account for URL endpoints. Even if there are multiple disks per node. Results below:

```
2019-01-04 17:51:16.132582 I | 172.17.0.3:9000's cpu utilization avg: 0.10% min: 0.08% max: 0.11%                                                                    2019-01-04 17:51:16.132629 I | 172.17.0.2:9000's cpu utilization avg: 0.08% min: 0.07% max: 0.09%                                                                    2019-01-04 17:51:16.134001 I | 172.17.0.3:9000's mem utilization: 70 MiB                                                                                             2019-01-04 17:51:16.134015 I | 172.17.0.2:9000's mem utilization: 71 MiB
```